SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -70,9 +72,44 @@ internal EnumCacheManager(string enumTypeName, int startingIndex, IEnumerable<in
                 }
 #endif
             }
+            else if (enumTypeName.CompareTo("CraftTreeType") == 0)
+            {
+#if DEBUG
+                Logger.Log("DEBUG: Now loading known CraftTreeTypes...");
+#endif
+                int addedToBanList = 0;
+                var enumValues = Enum.GetValues(typeof(CraftTree.Type));
```

Comment:
I'm pretty sure custom enum values aren't included in Enum.GetValues.

Reply:
[SMLHelper] DEBUG: Now loading known CraftTreeTypes...
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[0]
[SMLHelper] DEBUG: CraftTreeType ID=[0] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[1]
[SMLHelper] DEBUG: CraftTreeType ID=[1] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[2]
[SMLHelper] DEBUG: CraftTreeType ID=[2] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[3]
[SMLHelper] DEBUG: CraftTreeType ID=[3] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[4]
[SMLHelper] DEBUG: CraftTreeType ID=[4] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[5]
[SMLHelper] DEBUG: CraftTreeType ID=[5] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[6]
[SMLHelper] DEBUG: CraftTreeType ID=[6] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[7]
[SMLHelper] DEBUG: CraftTreeType ID=[7] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[8]
[SMLHelper] DEBUG: CraftTreeType ID=[8] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[9]
[SMLHelper] DEBUG: CraftTreeType ID=[9] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[10]
[SMLHelper] DEBUG: CraftTreeType ID=[10] is one of the game CraftTreeType.
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[48]
[SMLHelper] DEBUG: Found known CraftTreeType: ID=[49]
[SMLHelper] DEBUG: Finished known CraftTreeType exclusion. 2 ID were added in ban list.

Reply:
Huh. Didn't know.